cli/validation.py

```diff
@@ -0,0 +1,37 @@
+import click
+import re
+from uuid import UUID
+
+from cli.utils import RG_NAME_PATTERN
+from cli.utils import LOCATION_PATTERN
+
+def validate_uuid(context, param, value):
+    if is_uuid(value):
+        return value
+    else:
+        raise click.BadParameter('Subscription id must be in uuid4 format.')
+
+def is_uuid(param):
+    uuid_str = str(param)
+    try:
+        uuid_obj = UUID(uuid_str, version=4)
+    except ValueError:
+        return False
+    return str(uuid_obj) == uuid_str
+
+def validate_rg_name(context, param, value):
+    pattern = re.compile(RG_NAME_PATTERN)
+    if pattern.match(value):
+        return value
+    else:
+        raise click.BadParameter("""Resource group name should be 1-90
+        characters. Characters may be case insensitive, alphanumeric,
+        underscore, parentheses, hyphen, period (except at end), and Unicode characters.""")
```

Comment:
how does this print? is it one long line or does it print sensibly?

Reply:
Current: 
![image](https://user-images.githubusercontent.com/7232635/43652029-011b3840-9712-11e8-9e6c-640eae221450.png)

Without tabbed spacing:
![image](https://user-images.githubusercontent.com/7232635/43652096-3996bd2a-9712-11e8-876b-53b9e5416bad.png)

I'll do the latter.